src/components/ListResultTable/ListResultTable.jsx

```diff
@@ -1,59 +1,83 @@
-import {
-  useListController,
-  Loading,
-  ListBase,
-} from "react-admin";
+import {ListBase, Loading, useListController,} from "react-admin";
 import PropTypes from "prop-types";
-import { Component } from "react";
+import {Component, useEffect} from "react";
 import QueryResultList from "./QueryResultList/QueryResultList";
 
-/** 
+import {useLocation, useNavigate} from 'react-router-dom'
+
+/**
  * @param {object} props - the props passed down to the component
  * @returns {Component} custom List as defined by react-admin which either shows a loading indicator or the query results
  */
 function ListResultTable(props) {
-  const {
-    debounce,
-    disableSyncWithLocation,
-    exporter,
-    filter,
-    filterDefaultValues,
-    perPage,
-    resource,
-    sort,
-    ...rest
-  } = props;
-  const { isLoading } = useListController();
-  return (
-    <ListBase
-      debounce={debounce}
-      disableAuthentication={true} // A query can go over multiple sources, some of which he doesn't need authentication. Thus we don't know which need authentication and which do not. We still have authentication with this being true.
-      disableSyncWithLocation={disableSyncWithLocation}
-      exporter={exporter}
-      filter={filter}
-      filterDefaultValues={filterDefaultValues}
-      perPage={perPage}
-      queryOptions={{ keepPreviousData: false }}
-      resource={resource}
-      sort={sort}
-    >
-      {isLoading && <Loading />}
-      {!isLoading && <QueryResultList {...rest} />}
-    </ListBase>
-  );
+    const {
+        debounce,
+        disableSyncWithLocation,
+        exporter,
+        filter,
+        filterDefaultValues,
+        perPage,
+        resource,
+        sort,
+        variables,
+        ...rest
+    } = props;
+
+    const location = useLocation();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (variables) {
+            const queryParams = new URLSearchParams(location.search);
+
+            for (const [variableName, variableValue] of Object.entries(variables)) {
+                queryParams.set(variableName, variableValue)
+            }
+
+            navigate(
+                `?${queryParams.toString()}`
+            );
+        }
+    }, [location.search, navigate]);
```

Comment:
The history gets filled up with multiple instances of the exact same URL when loading a templated query with certain values for variables, so trying to go back in the browser's history results in returning to the same page for the templated query with the same values for variables multiple times.

Reply:
Better put this as comments in the code...

Reply:
Looking at the browser history in Firefox, it appears that there are only 2 entries for the last visited templated query: one without parameters in the URL (last but one entry) and one including parameters (last entry). When clicking the back arrow, it appears that the browser goes back to the last but one entry, but then immediately jumps forward again to the last entry, as if the above code is executed again.
See also my comment below about the UI experience - maybe to be solved all together.

Reply:
**Solved in 151b25b44aef96c994be9881d770e3fc2459419f**.